lib/location/location_core.c

```diff
@@ -281,6 +281,10 @@ int location_core_validate_params(const struct location_config *config)
 	}
 
 	for (int i = 0; i < config->methods_count; i++) {
+		if (config->methods[i].method == LOCATION_METHOD_WIFI_CELLULAR) {
```

Comment:
Method bitmask was something I thought but I admit that not for too long. Maybe I should have. I'm probably more reluctant to breaking changes than many others which is why I didn't think about it seriously enough. If I would do this from scratch, bitmask would be an obvious choice.

Bitmask has also its drawbacks as API user has some uncertainty which combinations can be used in input (method configuration) and output (event structure).

@tokangas What do you think?

Reply:
It did also cross my mind when reviewing the changes. It would have some benefits and it would allow flexibility in the future if there are more similar combined methods. On the other hand, there would be some ambiguity which combinations are valid.

Probably it wouldn't be a problem on the method configuration side, because even if the method IDs would be different, the overall structure of the configuration would remain the same. So, only one method bit would be allowed in one `location_method_config` and combining methods would still happen internally, right? And of course we could still use an enum in input and a bitmask in output, IDs for each method would just have to be declared twice.

With the current approach it's obvious which method combination(s) are valid, I guess that's the main benefit. I don't have a very strong opinion either way.

Reply:
After discussions today, we'd probably keep the enum. If we ever add a third method that could be combined to Wi-Fi and cellular, we'd need to move over to bitmask. However, we don't see the need for that in even in the next couple of years, but who knows...

Reply:
Any further comments @glarsennordic?

Reply:
@trantanen he might not respond anytime soon -- he is out sick.

Reply:
Ok, I'll request this PR to be merged so we get forward. You can add further comments here and we can continue discussion, and create another PR if necessary.

Reply:
Yep, sorry for my lack of response, thanks Pete for letting them know what happened to me.

Fair point, enum seems just fine for now. If you ever do add that third event, I would say this:

A third option would be to still use an enum for specifying desired request modes, but then use a separate set of bitmask definitions for specifying event sources, if you really want to avoid the bitmask for specifying request modes. Though IMO using a bitmask for everythign and just validating incoming bitmasks probably makes the most sense, yeah.

Reply:
> Though IMO using a bitmask for everythign and just validating incoming bitmasks probably makes the most sense, yeah.

Thinking only about the method specification, bitmask for the request mode would fine.
However, we need to take the method specific configuration into account and that will become a bit more unintuitive, or require changes to the structure that won't be perfect. So you'd need to make a compromise either in the configuration structures or not using the bitmask.
But this can be re-evaluated if there is a new method.